cmd/clusterctl/phases/pivot.go

```diff
@@ -25,28 +25,61 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/yaml"
 	"k8s.io/klog"
-	"sigs.k8s.io/cluster-api/cmd/clusterctl/clusterdeployer/clusterclient"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
-// Pivot deployes the provided provider components to a target cluster and then migrates
+type sourceClient interface {
+	Delete(string) error
+	DeleteMachineClass(namespace, name string) error
+	ForceDeleteCluster(string, string) error
+	ForceDeleteMachine(string, string) error
+	ForceDeleteMachineDeployment(string, string) error
+	ForceDeleteMachineSet(namespace, name string) error
+	GetClusterObjectsInNamespace(string) ([]*clusterv1.Cluster, error)
+	GetMachineClasses(string) ([]*clusterv1.MachineClass, error)
+	GetMachineDeployments(string) ([]*clusterv1.MachineDeployment, error)
+	GetMachineDeploymentsForCluster(string, string) ([]*clusterv1.MachineDeployment, error)
+	GetMachines(namespace string) ([]*clusterv1.Machine, error)
+	GetMachineSets(namespace string) ([]*clusterv1.MachineSet, error)
+	GetMachineSetsForCluster(string, string) ([]*clusterv1.MachineSet, error)
+	GetMachineSetsForMachineDeployment(*clusterv1.MachineDeployment) ([]*clusterv1.MachineSet, error)
+	GetMachinesForCluster(string, string) ([]*clusterv1.Machine, error)
+	GetMachinesForMachineSet(*clusterv1.MachineSet) ([]*clusterv1.Machine, error)
+	ScaleStatefulSet(string, string, int) error
+	WaitForClusterV1alpha1Ready() error
+}
+
+type targetClient interface {
```

Comment:
Would it make sense to push these interfaces up to clusterclient?

Reply:
I don't have a use case to push this up. If you do then we can definitely consider it!

For now though, I'm using this as a local scope down to make testing and readability of this functionality a bit easier